### ISSUE TITLE
Observations mismatch failes when one realization are missing the UNSMRY file.

### ISSUE THREAD

**Issue by kjetilbjorke** [equinor/fmu-ensemble#23]
Using the HistoryMatch container in webviz creates an EnsembleSet and mismatches from observations. The fmu.ensemble.Observations.mismatch failes when a realization has failed (does not have a summary file). 

https://github.com/equinor/webviz-subsurface/blob/2d90ffc138ace636c051f3a42e52dc12af586739/webviz_subsurface/datainput/_history_match.py#L30

https://github.com/equinor/fmu-ensemble/blob/0157bcfbad598a752324bb3904cd22a77546f566/src/fmu/ensemble/observations.py#L305

Traceback:
```
File "/.../venv/lib/python3.7/site-packages/webviz_subsurface/datainput/_history_match.py", line 30, in extract_mismatch
    .mismatch(ens_data)
  File "/.../venv/lib/python3.7/site-packages/fmu/ensemble/observations.py", line 116, in mismatch
    mismatches[(ensname, realidx)] = self._realization_mismatch(real)
  File "/.../venv/lib/python3.7/site-packages/fmu/ensemble/observations.py", line 305, in _realization_mismatch
    )[obsunit["key"]].values[0]
  File "/.../venv/lib/python3.7/site-packages/pandas/core/frame.py", line 2927, in __getitem__
    indexer = self.columns.get_loc(key)
  File "/.../venv/lib/python3.7/site-packages/pandas/core/indexes/base.py", line 2658, in get_loc
    return self._engine.get_loc(self._maybe_cast_indexer(key))
  File "pandas/_libs/index.pyx", line 108, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/index.pyx", line 132, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/hashtable_class_helper.pxi", line 1601, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File "pandas/_libs/hashtable_class_helper.pxi", line 1608, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: 'WBP4:SOME_WELL'
```

Using PR #3 does not help and gives this Traceback:
```
  File "/.../venv/lib/python3.7/site-packages/webviz_subsurface/datainput/_history_match.py", line 30, in extract_mismatch
    .mismatch(ens_data)
  File "/.../venv/lib/python3.7/site-packages/fmu/ensemble/observations.py", line 122, in mismatch
    mismatches[(ensname, realidx)] = self._realization_mismatch(real)
  File "/.../venv/lib/python3.7/site-packages/fmu/ensemble/observations.py", line 342, in _realization_mismatch
    )[obsunit["key"]].values[0]
  File "/.../venv/lib/python3.7/site-packages/pandas/core/frame.py", line 2927, in __getitem__
    indexer = self.columns.get_loc(key)
  File "/.../venv/lib/python3.7/site-packages/pandas/core/indexes/base.py", line 2658, in get_loc
    return self._engine.get_loc(self._maybe_cast_indexer(key))
  File "pandas/_libs/index.pyx", line 108, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/index.pyx", line 132, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/hashtable_class_helper.pxi", line 1601, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File "pandas/_libs/hashtable_class_helper.pxi", line 1608, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: 'WBP4:SOME_WELL'
```

**Comment by berland** [comment]
Possibly a Python 3 issue, seems to work in Python 2, see #25

**Comment by berland** [comment]
Nope, branch #3 with updated tests pass on Python 3.

In the obsfix branch, ValueErrors are captured at this position, but your trace reveals that your Python emits KeyError. Possibly the code should just capture both ValueError and KeyError and treat them the same, but we should understand it first.

**Comment by berland** [comment]
Is it so that you do have a summary file, but you do not have the same summary vectors in all of them?

**Comment by kjetilbjorke** [comment]
No the summary file is missing for this one realization (something crashed before simulations, hence only DATA file in eclipse/model folder).

**Comment by berland** [comment]
Hopefully fixed in a6d4ae (part of #3 )

**Comment by kjetilbjorke** [comment]
Works with #3 for me.
